src/primitives/gf32.rs

```diff
@@ -285,58 +256,43 @@ impl AsRef<u8> for Fe32 {
     fn as_ref(&self) -> &u8 { &self.0 }
 }
 
-/// Implements $op for the 2x2 matrix of type by ref to type
-macro_rules! impl_op_matrix {
-    ($op:ident, $op_fn:ident, $call_fn:ident) => {
-        impl ops::$op<Fe32> for Fe32 {
-            type Output = Fe32;
-            #[inline]
-            fn $op_fn(self, other: Fe32) -> Fe32 { self.$call_fn(other) }
-        }
-
-        impl ops::$op<Fe32> for &Fe32 {
-            type Output = Fe32;
-            #[inline]
-            fn $op_fn(self, other: Fe32) -> Fe32 { self.$call_fn(other) }
-        }
-
-        impl ops::$op<&Fe32> for Fe32 {
-            type Output = Fe32;
-            #[inline]
-            fn $op_fn(self, other: &Fe32) -> Fe32 { self.$call_fn(*other) }
+impl super::Field for Fe32 {
+    const ZERO: Self = Fe32::Q;
+    const ONE: Self = Fe32::P;
+    const GENERATOR: Self = Fe32::Z;
+    const MULTIPLICATIVE_ORDER: usize = 31;
+    const MULTIPLICATIVE_ORDER_FACTORS: &'static [usize] = &[1, 31];
+
+    fn _add(&self, other: &Fe32) -> Fe32 { Fe32(self.0 ^ other.0) }
+    // Subtraction is the same as addition in a characteristic-2 field
+    fn _sub(&self, other: &Fe32) -> Fe32 { Fe32(self.0 ^ other.0) }
```

Comment:
Why not call `_add` within the subtraction function to express the equivalence explicitly, rather than duplicating the implementation?